src/toggle/Toggle.tsx

```diff
@@ -0,0 +1,73 @@
+import "./_toggle.scss";
+
+import React, {createContext, useState} from "react";
+import classNames from "classnames";
+
+import ToggleItem from "./item/ToggleItem";
+
+export interface ToggleItemShape {
+  id: string;
+  children: React.ReactNode;
+}
+
+const ToggleContext = createContext({
+  selectedToggleItemsState: [] as ToggleItemShape[],
+  setSelectedToggleItemsState: (() => null) as React.Dispatch<
+    React.SetStateAction<ToggleItemShape[]>
+  >,
+  isMultiple: false,
+  onToggleItem: (id: string) => id
+});
+
+export interface ToggleProps {
+  children: React.ReactNode;
+  onToggle: (id: string) => void;
+  isMultiple?: boolean;
+  position?: "vertical" | "horizontal";
+  isDisabled?: boolean;
+  customClassName?: string;
+  testid?: string;
+}
+
+function Toggle({
+  children,
+  onToggle,
+  isMultiple = false,
+  position = "horizontal",
+  isDisabled,
+  customClassName,
+  testid
+}: ToggleProps) {
+  const toggleClassName = classNames("toggle", customClassName, {
+    "toggle--is-horizontal": position === "horizontal",
+    "toggle--is-vertical": position === "vertical",
+    "toggle--is-disabled": isDisabled
+  });
+  const [selectedToggleItemsState, setSelectedToggleItemsState] = useState<
+    ToggleItemShape[]
+  >([]);
+
+  return (
+    <div data-testid={testid} className={toggleClassName}>
+      <ToggleContext.Provider
+        value={{
+          selectedToggleItemsState,
+          setSelectedToggleItemsState,
+          isMultiple,
+          onToggleItem
+        }}>
+        {children}
+      </ToggleContext.Provider>
+    </div>
+  );
+
+  function onToggleItem(id: string) {
+    onToggle(id);
+
+    return id;
+  }
```

Comment:
Why does this function returns `id`? I couldn't figure out the use case here

Reply:
I thought we could access the item `id` that was toggled. `onToggleItem` function handle both of `ToggleItem` and the parent components handle function. But if it's confusing, we can think of an easier usage 🤔 

```
 onToggleItem: (id: string) => id
```

Reply:
But we already pass `id` to `onToggle`, so the parent will know the `id`. Am I missing something ? 🤔

Reply:
While the context creating, `onToggleItem` takes an `id` parameter and returns this. Because `Toggle` should know clicked item for getting action. In the don't return `id` case, we can't handle specific `onToggle` action.
If you know another easier way, please share with me :)

Reply:
I see now, thanks 👍